File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.cpp
class Solution {
public:

    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        int n = words.size();
        vector<int> pref(n+1, 0);

        unordered_set<char> vow = {'a', 'e', 'i','o','u'};

        for(int i=0; i<n;i++){
            pref[i+1] = pref[i];
            if(vow.count(words[i].front()) && 
                vow.count(words[i].back())){
                    pref[i+1]++;
            }
            
        }

        vector<int> ans;

        for(auto it : queries){
            int l = it[0];
            int r = it[1];

            ans.push_back(pref[r+1] - pref[l]);

        }

        return ans;
    }
};

File: 2691-count-vowel-strings-in-ranges/2691-count-vowel-strings-in-ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        n = len(words)
        pref = [0] * (n + 1)
        ans = []

        vowels = {'a', 'e', 'i', 'o', 'u'}

        for i in range(n):
            pref[i+1] = pref[i]

            if words[i][0] in vowels and words[i][-1] in vowels:
                pref[i+1] += 1
            
        for q in queries:
            left = q[0]
            right = q[1]

            ans.append(pref[right + 1] - pref[left])
        
        return ans